psci/Main.hs

```diff
@@ -293,21 +299,21 @@ browserBackend serverPort = Backend setup evaluate reload shutdown
       result <- takeMVar resultVar
       putStrLn result
 
-nodeBackend :: [String] -> Backend
-nodeBackend nodeArgs = Backend setup eval reload shutdown
+nodeBackend :: Maybe FilePath -> [String] -> Backend
+nodeBackend nodePath nodeArgs = Backend setup eval reload shutdown
   where
     setup :: IO ()
     setup = return ()
 
     eval :: () -> String -> IO ()
     eval _ _ = do
       writeFile indexFile "require('$PSCI')['$main']();"
-      process <- findNodeProcess
+      process <- (nodePath <|>) <$> findNodeProcess
```

Comment:
Bit of a nit, but I think I'd prefer `process <- maybe findNodeProcess pure nodePath` here, so that `findNodeProcess` doesn't get run in the case where `nodePath` is provided. That way we can be sure that `findNodeProcess` isn't going to do anything dodgy like write to the console or throw an error in a case where we don't even need it because a `nodePath` has been provided, without having to look at its definition.